File: food-restaurant/src/main/kotlin/com/food/restaurant/gateway/database/model/RestaurantModel.kt
package com.food.restaurant.gateway.database.model

class RestaurantModel(
        val id: Long?,
        val name: String,
        val category: CategoryModel,
        val logo: String,
        val description: String,
        val address: String
) {
    constructor(name: String, category: CategoryModel, logo: String,
                description: String, address: String) :
            this(null, name, category, logo, description, address)
}

File: food-ui-webapp/src/services/Restaurant.model.ts
export default class Restaurant {
    _uuid: string;
    _name: string;
    _logo: string;
    _description: string;
    _address: string;
  
    constructor(uuid: string, name: string, logo: string, description: string, 
                address: string) {
      this._uuid = uuid;
      this._name = name;
      this._logo = logo;
      this._description = description;
      this._address = address;
    }
  
    get uuid(): string {
      return this._uuid;
    }
  
    get name(): string {
      return this._name;
    }
  
    get logo(): string {
      return this._logo;
    }
  
    get description(): string {
      return this._description;
    }
  
    get address(): string {
      return this._address;
    }
  }